File: API/Features/Reservations/Registrars/Dtos/RegistrarReadDto.cs
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Reservations.Registrars {

    public class RegistrarReadDto : IBaseEntity, IMetadata {

        // PK
        public int Id { get; set; }
        // Fields
        public string Fullname { get; set; }
        public string Phones { get; set; }
        public string Email { get; set; }
        public string Fax { get; set; }
        public string Address { get; set; }
        public bool IsPrimary { get; set; }
        public bool IsActive { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }
        // Navigation
        public SimpleEntity Ship { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/registrars/classes/dtos/registrar-read-dto.ts
import { Metadata } from 'src/app/shared/classes/metadata'
import { ShipAutoCompleteVM } from '../../../ships/classes/view-models/ship-autocomplete-vm'

export interface RegistrarReadDto extends Metadata {

    // PK
    id: number
    // Object fields
    ship: ShipAutoCompleteVM
    // Fields
    fullname: string
    phones: string
    email: string
    fax: string
    address: string
    isPrimary: boolean
    isActive: boolean

}
